src/examples/cpp/shld_test.cpp

```diff
@@ -0,0 +1,166 @@
+
```

Comment:
Thanks to provided tests but i'm using this [file](https://github.com/JonathanSalwan/Triton/blob/master/src/samples/ir_test_suite/ir.c) for my semantics tests and also the `qemu-test-x86_64`.